pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,222 @@
+
+	pvcExample = templates.Examples(i18n.T(`
+	    # Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode and "1Gi" capacity.
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi
+
+		# Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode, "1Gi" capacity and a storage class named "HighQuality".
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi --storage-class=HightQuality
+
+		# Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode, "1Gi" capacity, requiring a persistent volume named "HighQualityVolume".
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi --volume-name=HightQualityVolume
+
+		# Create a persistent volume claim named "myclaim" with multiple access modes and "1Gi" capacity.
+		kubectl create persistentvolumeclaim myclaim --access-mode=RWO,RWX --capacity=1Gi`))
+)
+
+type CreatePVCOptions struct {
+	Name         string
+	AccessModes  []api.PersistentVolumeAccessMode
+	Capacity     *resource.Quantity
+	StorageClass *string
+	VolumeName   string
+
+	DryRun       bool
+	OutputFormat string
+	Namespace    string
+	Client       internalversionstorage.PersistentVolumeClaimInterface
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+// PersistentVolumeClaim is a command to ease creating PersistentVolumeClaims.
+func NewCmdCreatePersistentVolumeClaim(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePVCOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaim NAME --access-mode=accessmode --capacity=capacity [--storage-class=storageclass] [--volume-name=volumename] [--dry-run]",
+		Short:   pvcLong,
+		Long:    pvcLong,
+		Example: pvcExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePVC())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	cmd.Flags().StringSlice("access-mode", []string{}, "AccessMode contains the desired access modes the volume should have.")
+	cmd.Flags().String("capacity", "", "Capacity represents the minimum storage capacity required.")
+	cmd.Flags().String("storage-class", "", "Name of the StorageClass required by the claim.")
+	cmd.Flags().StringVar(&c.VolumeName, "volume-name", "", "VolumeName is the binding reference to the PersistentVolume backing this claim.")
+
+	return cmd
+}
+
+func (c *CreatePVCOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+
+	// Transform access mode abbreviation to complete access mode and remove duplicate access mode.
```

Comment:
not sure we must do this in client-side, but seems have mary command  not remove duplicate?

Reply:
IMO, it's better to do the validate and duplicate remove work as much as possible from client side. Other reviewers, your opinions?

Reply:
I think we'd better leave this to apiserver, not on client side. Since validations from apiserver is a must and the only gate for all requests, there is no need to introduce duplicate validation.

Reply:
yes, but the current behavior is that the api server does not check enough.
e.g. if we try to create a PVC with duplicate access mode like:
```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: hostpath
spec:
  accessModes:
    - ReadWriteMany
    - ReadWriteMany
  resources:
    requests:
      storage: 1Mi
```
the above PVC can be successfully created and when we get the PVC, we can still see the duplicate access mode(I believe this is a case for some other resources in K8s).

IIUC, we'd better avoid sending request to server side before the validation passes, so some of the validation should be done before we are trying to create the resource from API server side.

Reply:
> the current behavior is that the api server does not check enough.

@xingzhou Then we should enhance in api validation.

> IIUC, we'd better avoid sending request to server side before the validation passes, so some of the validation should be done before we are trying to create the resource from API server side.

What will happen if we send the same request data using pure REST api requests? No validation will be performed?

To keep it consistent, I'd prefer to make api validation as the only gate keeper. Then no matter we send the request using CLI or REST, the same validation will be performed.

Reply:
Also we're making `kubectl` much more simpler and friendly, such as fetching the valid resources from apiserver instead of current hard-coded way. IMO, we should not add such an extra burden to `kubectl`.

Reply:
For duplicate check, I'm afraid there is no such validation.

It's fine to me to remove the check duplicate part and access mode validation here in the kubectl side. But for the following part, still need to be kept:
"Transform the abbreviations to complete access mode(as RWO/RWX and ROX are CLI stuff, not recognized by api server)"
I'll update the patch then